### PR TITLE
update search highlights on page mount if results already populated

### PR DESCRIPTION
**Pull request by bethshook**
for: https://app.zenhub.com/workspaces/openstax-unified-5b71aabe3815ff014b102258/issues/openstax/unified/1743

### CODE REVIEW

File: src/app/content/components/Page/PageComponent.tsx [hunk 1]
```diff
@@ -66,6 +66,13 @@ export default class PageComponent extends Component<PagePropTypes> {
     // tslint:disable-next-line: max-line-length
     this.highlightManager = highlightManager(this.container.current, () => this.props.highlights, this.props.services, this.props.intl);
     this.scrollToTopOrHashManager = scrollToTopOrHashManager(this.container.current);
+
+    if (this.props.searchHighlights.selectedResult) {
+      this.searchHighlightManager.update({searchResults: [], selectedResult: null}, this.props.searchHighlights, {
```
Comment (TomWoodward) [inline src/app/content/components/Page/PageComponent.tsx:71]: maybe we should tweak the manager to allow the prev structure to just be null

**Review by TomWoodward — CHANGES_REQUESTED** [review]
other than that one comment this looks fine

File: src/app/content/components/Page/PageComponent.tsx (hunk 1 above)
Comment (bethshook) [reply src/app/content/components/Page/PageComponent.tsx:71]: this is the update handler so I don't know if it's worth tweaking:
```
const handleUpdate = (services: Services) => (
  previous: HighlightProp,
  current: HighlightProp,
  options: UpdateOptions
) => {
  updateResults(services, previous, current, options);
  selectResult(services, previous, current, options);
};
```

Comment (TomWoodward) [reply src/app/content/components/Page/PageComponent.tsx:71]: yea you'd have to change it to be nullable all the way down, which sucks, but also manually constructing the empty state here is pretty jank, so i think that might be a good option

**Review by TomWoodward — APPROVED** [review]

**Pull request merged**